src/identity/contactMapping.ts

```diff
@@ -411,3 +419,64 @@ export function getAddressFromPhoneNumber(
   // Normal case when there is only one address in the mapping
   return addresses[0]
 }
+
+function calculateHash(str: string) {
+  const hash = crypto.createHash('sha256')
+  hash.update(str)
+  return hash.digest('hex')
+}
+
+export function* saveContacts() {
+  try {
+    const saveContactsGate = getFeatureGate(StatsigFeatureGates.SAVE_CONTACTS)
+    const phoneVerified = yield* select(phoneNumberVerifiedSelector)
+    // TODO(satish): use rn permissions
```

Comment:
will do this in a followup as some refactoring is needed in the select recipient screen to share this code.